### DISCUSSION TITLE
Is it possible to target files in node_modules in inputs/outputs?

### DISCUSSION THREAD

**Discussion by KidkArolis** [vercel/turborepo#5122] [Help] [answered]
When using ReScript and running `rescript build -with-deps` it compiles files in node_modules. However with `turbo` I couldn't get it to recognise those relevant files as input or output:

With this `turbo.json` config:

```
{
  "extends": ["//"],
  "pipeline": {
    "build": {
      "outputs": [
        "lib/bs/**/*",
        "src/__generated__/**/*",
        "src/**/*.bs.js",
        "node_modules/**/*.bs.js"
      ],
      "inputs": [
        "src/**/*",
        "node_modules/**/*.{res,ml,mli,atd}",
        "bsconfig.json",
        "relay.config.js"
      ]
    }
}
```

Running `turbo build` will compile

```
packages/client/node_modules/@ahrefs/bs-atdgen-codec-runtime/src/atdgen_codec_decode.ml
```

to

```
packages/client/node_modules/@ahrefs/bs-atdgen-codec-runtime/src/atdgen_codec_decode.bs.js
```

But then if I delete `packages/client/node_modules/@ahrefs/bs-atdgen-codec-runtime/src/atdgen_codec_decode.bs.js`, and run `turbo build` again, turbo says "cache hit" and does not regenerate the missing `.bs.js` file.

**Comment by anthonyshew** [comment, marked as answer]
There is no way to do this in Turborepo and I'm not sure that we would want to introduce this feature. There are many, many ways to shoot oneself in the foot with this.

I would recommend not caching that task [using `"cache": false`](https://turbo.build/repo/docs/reference/configuration#cache).
